lib/resty/core/base.lua

```diff
@@ -248,10 +248,7 @@ do
     ok, exdata = pcall(require, "thread.exdata")
     if ok and exdata then
         function _M.get_request()
-            local r = exdata()
-            if r ~= nil then
```

Comment:
The guard here is used to convert `cdata:NULL` to a Lua `nil`.

Reply:
then we may use  `local r = exdata` for it to convert and then directly `return r`